src/components/file-dropzone/file-dropzone.tsx

```diff
@@ -0,0 +1,162 @@
+import {
+    Component,
+    h,
+    Event,
+    EventEmitter,
+    Host,
+    Prop,
+    State,
+} from '@stencil/core';
+import { FileInfo } from '../../global/shared-types/file.types';
+import { createFileInfo, isTypeAccepted } from '../../util/files';
+
+/**
+ * This component enables you to seamlessly convert any region of the user interface into
+ * a file dropzone area, just by wrapping it inside the `limel-file-dropzone`.
+ *
+ * The file dropzone can then be used to allow end-users to upload files
+ * by dragging and dropping them into the specified area.
+ *
+ * After receiving the files, the component emits a `filesSelecgted` event. For unsupported
+ * files (specified with the `accept` prop) a `filesRejected` event will be emitted.
+ *
+ * The event detail would be an array of `FileInfo` objects,
+ * each representing a file dropped into the dropzone.
+ *
+ * @exampleComponent limel-example-file-dropzone
+ * @exampleComponent limel-example-file-dropzone-type-filtering
+ * @private
+ */
+@Component({
+    tag: 'limel-file-dropzone',
+    shadow: true,
+    styleUrl: 'file-dropzone.scss',
+})
+export class FileDropzone {
+    /**
+     * Specifies the types of files that the dropzone will accept. By default, all file types are accepted.
+     *
+     * For media files, formats can be specified using: `audio/*`, `video/*`, `image/*`.
+     * Unique file type specifiers can also be used, for example: `.jpg`, `.pdf`.
+     * A comma-separated list of file extensions or MIME types is also acceptable, e.g., `image/png, image/jpeg` or
+     * `.png, .jpg, .jpeg`.
+     *
+     * @see [HTML attribute: accept](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/accept) for more
+     * details.
+     */
+    @Prop({ reflect: true })
+    public accept: string = '*';
+
+    /**
+     * Set to `true` to disable the file dropzone.
+     */
+    @Prop()
+    public disabled: boolean = false;
+
+    /**
+     * Text to be displayed when the user is dragging a file over the dropzone.
+     */
+    @Prop()
+    public onDragLayoutText: string = '';
```

Comment:
this is a really strange name for a public prop 😅

```suggestion
    public label: string = '';
```

or 

```suggestion
    public text: string = '';
```

Reply:
In most of our components we use these two prop names. I prefer `text` myself. Because we wanna add another prop, preferably called `helperText`, which is something to display below this text.

<img width="416" alt="image" src="https://github.com/Lundalogik/lime-elements/assets/35954987/c32989eb-b582-429f-bb5b-c8aa5e6557fb">

Reply:
I left one icon 😢, and  will change it when we have a better one